### ISSUE TITLE
Bring back the phase argument for lightcurve.fold() [Redux of Issues #357 and #361]

### ISSUE THREAD

**Issue by KenMighell** [KeplerGO/lightkurve#384]
#### Problem description

The lightcurve.fold() method used to have an optional phase argument.  Recently following disccusions in Issues #357 & #361, the phase argument was removed and an optional transit_midpoint argument was added.  

The removal of the phase argument broke one of my demos for the upcoming American Astronomical Society meeting in Seattle.  I partially fixed the demo by commenting out the use of the phase argument; the fix was only partial as the resulting plots do not look as good as before.

This raises the issue that **not all periodic variable objects are transiting exoplanets**.  

Phase was useful in modifying (hacking) lightkurve phase plots of RR Lyrae variables (or eclipsing binaries) produced by the lightcurve.fold() method so that the minimum flux had a phase value of zero -- this is a current convention among astronomers doing research with variable stars.  This was also useful during the production of quick-look (non-publication grade) phase plots used during data exploration.

Suggestions:

[1]  Adding phase=None to the definition of the fold() method in lightcurve.py combined with the detection of the None value doing *nothing* would enable current applications which use transit_midpoint to continue working as expected.  

[2]  If a phase argument *is* used with a float value between -0.5 and +0.5, then the phase plot X-axis data could be shifted by the requested phase value (as before) and older lightkurve apps using phase would continue working as expected.  

[3]  If *both* phase and transit_midpoint were to be defined, then the phase value would be ignored.

[4]  If phase *is* added back,  the old docstring describing the use of phase would need to be improved (clarified).

#### Environment:
- platform (e.g. Linux, OSX, Windows): Mac OS X (Sierra and El Capitan)
- lightkurve version: 1.0b26.dev
- installation method: git

**Comment by barentsen** [comment]
Thanks for reporting this @KenMighell.

I'm sorry to hear the change broke your demo.  The meaning of the `phase` parameter was indeed very unclear in the past, which led to change.

For the use case you describe, perhaps a new parameter called `phase_shift` could be introduced, which would allow the phase to be shifted by some amount.  It would default to zero and be applied in addition to `transit_midpoint`.

However, we would want to be certain that `phase_shift` adds functionality that is necessary and important.  Right now one can also hack the `transit_midpoint` parameter to plot a folded RR Lyrae star with minimum flux at phase zero roughly as follows:
```Python
lc.fold(period=xxx, transit_midpoint=lc.time[np.argmin(lc.flux)])
```

Admittedly, for RR Lyrae stars it is annoying to have this parameter called `transit_midpoint`.  It may be more appropriate to call this parameter eg. `phase0_time`.

Thoughts, comments, anyone?

**Comment by KenMighell** [comment]
> On Dec 28, 2018, at 6:33 AM, Geert Barentsen <notifications@github.com> wrote:
> 
> Thanks for reporting this @KenMighell <https://github.com/KenMighell>.
> 
> I'm sorry to hear the change broke your demo. The meaning of the phase parameter was indeed very unclear in the past, which led to change.
> 
> For the use case you describe, perhaps a new parameter called phase_shift could be introduced, which would allow the phase to be shifted by some amount. It would default to zero and be applied in addition to transit_midpoint.  
> 
This is exactly the behavior I was trying to describe.  phase_shift is a better (more descriptive) name for the argument.
> However, we would want to be certain that phase_shift adds functionality that is necessary and important. Right now one can also hack the transit_midpoint parameter to plot a folded RR Lyrae star with minimum flux at phase zero roughly as follows:
> 
> lc.fold(period=xxx, transit_midpoint=lc.time[np.argmin(lc.flux)])
I do think that phase_shift would add useful functionality.   The above code may not be obvious for a new 
lightkurve user who is simultaneously learning Python (e.g., a bright undergraduate).   Maybe *both* could 
be used in an official introductory lightkurve tutorial showing how to produce phased light curves from K2 
observations?
> Admittedly, for RR Lyrae stars it is annoying to have this parameter called transit_midpoint. It may be more appropriate to call this parameter eg. phase0_time.
> 
I vote for phase0_time as being more general.

**Comment by ernewton** [comment]
The docs page http://docs.lightkurve.org/tutorials/how-to-recover-the-first-tess-candidate.html still has phase instead of midpoint in the TESS example, I presume because I'm one version ahead of the docs.

**Comment by gully** [comment]
Indeed, calling it transit_midpoint is a bit of a misnomer, and sometimes it's convenient to specify in radians or other transformations.  I'm updating the tutorials to reflect recent code changes, so the docs should be up-to-date momentarily.

**Comment by MariannaPat** [comment]
Hello, I am currently working on exoplanet lightcurves and I would like to ask if there is a way to shift the phase (x-axis) from (-0.5, 05) to (0.0,1.0).
